src/main/java/org/apache/sling/api/redirect/RedirectResponse.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.sling.api.redirect;
+
+/**
+ * <p>
+ * A Redirect redirect defines the base class used to hold information to build a redirect.  It is extended
+ * by the caller to RedirectResolver.resolve, and called by the implementation of the RedirectResolver interface.
+ * On return from that call the caller will use the information in the RedirectResponse to perform whatever operation
+ * it needs to perform. Typically this will be returning a http redirect, but there may be other uses.
+ * </p><p>
+ * A class is being used rather than an interface to allow empty setter methods to be added over time.
+ * Callers to RedirectResolver.resolve should override those methods that they need information for. New methods to this
+ * class should be added with empty implementations to ensure that pre-exiting classes extending this method to not
+ * have to be updated.
+ * </p>
+ */
+public class RedirectResponse {
```

Comment:
I rather would make this a provider type interface (which if enhanced would only need an update to the get servlet providing an implementation) - or an abstract class. But a public class with empty methods looks really strange

Reply:
With an interface, this package is no better than the URIProvider, since every implementation of the RedirectResponse will break and require an update if RedirectResponse changes. Sling Default GET is not the only potential location of an implementation. Every bundle that uses the RedirectResolver, will need to implement this class.   Using  concrete non-final class means that neither side of the caller and called has to be updated as the class changes, except where the caller or called needs to leverage additional functionality. It may be a unusual pattern, and interface or abstract will only create the same problem that was experienced with URIProvider.

Reply:
fixed empty methods as recommended by Sonar

Reply:
well, interfaces have default methods since some time, an abstract class can be extended without breaking implementors etc. But if you wish to go with what yo have, than lets to that. But it stays strange

Reply:
I'm -1 on this class like it is. We should not expose non-final classes as API. Moreover, by not providing any versioning annotation, this class will be a `Consumer` type, which means that we expect a lot of implementations.

What I also don't like about the class (and an interface with the same methods would suffer from the same issues) is that it provides only setters as API, when an actual caller will be interested in the data this object stores. If the class acts only as a DTO, then I'd make it provide only getters. Additional getters will indeed affect implementations, but only if the interface doesn't provide `default` methods, which is possible since Java 8. Consumers like the default GET servlet, however, will not be affected.

Another method I'd add is the `shouldRedirect` - I think it's more obvious to call this method first, rather than making the caller check first that the `RedirectResponse` has a URL and a status code different than the defined `NO_REDIRECT`.

Reply:
I dont see how we can make this work as an API and not cause disruption if it ever needs to change. I will look for alternative methods of doing this.

I understand and respect the -1.

Reply:
moved to an interface and added attributes to address the need to be able to make unforeseen changes in production deployments.